### ISSUE TITLE
"follow" a growing file

### ISSUE THREAD

**Issue by dtkerns** [rshk/python-pcapng#16]
I'm using dumpcap to write a file. I have been unsuccessful in getting python-pcapng to "follow" the growing file (it races to the "current" end-of-file and stops) Please provide the capability (or an example if I'm just missing how to) pause for more data rather than assume EOF.

Ideally, dumpcap would write to a named pipe (fifo), and python-pcapng would read the fifo but dumpcap refuses to do that.

edit: I've since learned that more modern versions of dumpcap *can* write to both a pipe or a named pipe.

**Comment by rshk** [comment]
I think the best option here, if you want to keep streaming new content from a file (`tail -f` style), is to have an object that acts like a normal stream (with a `read()` method), but underneath just blocks on EOF and waits for new data to come in from the file, rather than returning.
You can then pass such an object to the library to achieve what you were asking.

**Comment by dtkerns** [comment]
Yes, that's what I was after... my intended usage was:
`
dumpcap -pq -i em1 -f $filter -w - | mypcapng_reader | myprotocol_analyser
`
what I've ended up using was this:
`
tcpdump -pnnqUxt -i em1 $filter | mypcap_reader | myprotocol_analyser
`
so the urgency is gone. But I still think it's a worthwhile feature.

**Comment by rshk** [comment]
@dtkerns I don't think this is something that belongs into python-pcapng itself, but rather something the caller can use to achieve the desired effect.

I.e. the library interface is to "accept a stream object from which to read data"; providing a stream that listens to changes to the file is up to the caller (as there are simply too many possible scenarios for data streaming needs, and streaming data is beyond the scope of the library).

So basically something like what described here https://stackoverflow.com/questions/1703640/how-to-implement-a-pythonic-equivalent-of-tail-f should work just fine to feeding data into the parser.
(Just keep in mind the accepted answer is about handling line-based text files, so it needs to be adapted for handling binary data instead).
Hopefully there might already be a library somewhere to do this, without need to reimplement it yourself.
